Xamarin.Forms.Core/MultiBinding.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms
+{
+	[ContentProperty(nameof(Bindings))]
+	public sealed class MultiBinding : BindingBase
+	{
+		IMultiValueConverter _converter;
+		object _converterParameter;
+		BindingExpression _expression;
+		Collection<BindingBase> _bindings;
+		BindableProperty _targetProperty;
+		bool _isApplying;
+		bool _isCreating;
+		bool _hasSuccessfullyConverted;
+
+		public IMultiValueConverter Converter
+		{
+			get { return _converter; }
+			set
+			{
+				ThrowIfApplied();
+				_converter = value;
+			}
+		}
+
+		public object ConverterParameter
+		{
+			get { return _converterParameter; }
+			set
+			{
+				ThrowIfApplied();
+				_converterParameter = value;
+			}
+		}
+
+		public Collection<BindingBase> Bindings
+		{
+			get => _bindings ?? (_bindings = new Collection<BindingBase>());
+			set => _bindings = value;
```

Comment:
it should probably throw if applied, and also throw if the collection is modified once the binding is applied

Reply:
Ok on throw if applied. To detect collection modified wouldn't it have to be an INCC though? Is it worth it?

Reply:
> I know there's none in WPF, but what would be a good markup extension for this ? (we can bring that on a different PR)

I was thinking about that myself, but I don't know of any precedent for a collection being represented through a markup extension, do you?